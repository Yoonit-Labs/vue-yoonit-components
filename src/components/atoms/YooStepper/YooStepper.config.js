module.exports = {
  counterType: {
    options: [
      'dot',
      'number'
    ]
  }
}
