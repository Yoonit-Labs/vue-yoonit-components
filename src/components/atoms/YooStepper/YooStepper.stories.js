import yooStepper from './YooStepper.vue'
import PropsConfig from './YooStepper.config'

export default {
  title: 'Yoonit/Stepper',
  component: yooStepper,
  argTypes: {
    counterType: {
      control: {
        type: 'select',
        options: PropsConfig.counterType.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooStepper },
  template:
    '<yoo-stepper v-bind="$props"></yoo-stepper>'
})

export const Custom = Template.bind({})
Custom.args = { totalSteps: 5, counterType: 'number' }

export const Dots = Template.bind({})
Dots.args = { totalSteps: 5, counterType: 'dot' }

export const Number = Template.bind({})
Number.args = { totalSteps: 3, counterType: 'number' }
