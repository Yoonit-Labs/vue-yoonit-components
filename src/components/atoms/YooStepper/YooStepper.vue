<template lang="pug">
YooFlexLayout.yoo-stepper(
  flexDirection="row"
  justifyContent="space-around"
  alignItems="center"
)
  div(
    v-if="counterType === 'number'"
    class="yoo-stepper__number m__t--xs"
  ) {{takeNumberLabel}}
  div(
    v-if="counterType === 'dot'"
    v-for="(step, index) in takeStepsLength"
    :key="'dotStep-' + index"
    class="yoo-stepper__dot m__r--l"
    :class=`[
      { 'yoo-stepper__dot--active': index === currentStep },
      { 'yoo-stepper__dot--last': index === totalSteps }]`
  )
</template>

<script>

import YooFlexLayout from '@/components/quarks/YooFlexLayout/YooFlexLayout.vue'

export default {
  name: 'YooStepper',
  components: {
    YooFlexLayout
  },
  props: {
    counterType: {
      type: String,
      default: 'dot',
      validator (value) {
        return [
          'dot',
          'number'
        ].indexOf(value) !== -1
      }
    },
    totalSteps: {
      type: Number,
      default: 1
    },
    currentStep: {
      type: Number,
      default: 0
    }
  },
  computed: {
    /**
    * @description Returns an array of the totalSteps prop
    * @computed takeStepsLength
    * @returns {array}
    */
    takeStepsLength () {
      return new Array(this.totalSteps)
    },
    /**
    * @description Returns the currentStep prop adding 1
    * @computed takeCurrentStepNumber
    * @returns {number}
    */
    takeCurrentStepNumber () {
      return this.currentStep + 1
    },
    /**
    * @description Returns a formatted string when the counterType prop is equal to type "number"
    * @computed takeNumberLabel
    * @returns {string}
    */
    takeNumberLabel () {
      return `${this.takeCurrentStepNumber} de ${this.totalSteps}`
    }
  }
}
</script>

<style src="./YooStepper.sass" lang="sass" scoped></style>
