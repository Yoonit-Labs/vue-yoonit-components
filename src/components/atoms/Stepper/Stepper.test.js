/**
 * ██╗   ██╗ ██████╗  ██████╗ ███╗   ██╗██╗████████╗
 * ╚██╗ ██╔╝██╔═══██╗██╔═══██╗████╗  ██║██║╚══██╔══╝
 *  ╚████╔╝ ██║   ██║██║   ██║██╔██╗ ██║██║   ██║
 *   ╚██╔╝  ██║   ██║██║   ██║██║╚██╗██║██║   ██║
 *    ██║   ╚██████╔╝╚██████╔╝██║ ╚████║██║   ██║
 *    ╚═╝    ╚═════╝  ╚═════╝ ╚═╝  ╚═══╝╚═╝   ╚═╝
 *
 * https://yoonit.dev - about@yoonit.dev
 *
 * Vue Yoonit Components
 * VueJS Atomic Design System framework
 *
 * Vitória Costa, Tiago Brito, Fernando Junior, Sabrina Sampaio, Gabriel Mule, Gabriel Moraes, Gabriel Rizzo & Luigui Delyer @ 2020-2021
 */

import { shallowMount } from '@vue/test-utils'
import YooStepper from '@/components/atoms/Stepper/Stepper.vue'
import PropsConfig from '@/components/atoms/Stepper/Stepper.config'

const classBlock = 'yoo-stepper'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooStepper, {
    slots: { default: SlotText }
  })
}

describe('YooStepper Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-stepper').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('counterType', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.counterType.options.includes(YooStepper.props.counterType.default)).toBe(true)
      })
      PropsConfig.counterType.options.forEach(counterType => {
        it(`Includes variation class: .${classBlock}__${counterType}`, async () => {
          await wrapper.setProps({ counterType })
          expect(wrapper.find(`.${classBlock}__${counterType}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
})
