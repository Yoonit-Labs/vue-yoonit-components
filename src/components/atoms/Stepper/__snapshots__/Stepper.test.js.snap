// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooStepper Component Matches Snapshot 1`] = `
<yooflexlayout-stub flexdirection="row" flexwrap="nowrap" justifycontent="space-around" alignitems="center" aligncontent="stretch" gap="0" class="yoo-stepper">
  <!---->
  <div class="yoo-stepper__dot m__r--m m__l--m yoo-stepper__dot--active"></div>
</yooflexlayout-stub>
`;
